modules/xmpp/JingleSessionPC.js

```diff
@@ -1508,61 +1650,48 @@ export default class JingleSessionPC extends JingleSession {
         }
-        const workFunction = () => {
-            if (!this.peerconnection) {
-                return false;
+        const queueTask = new ModificationQueueTask(thisTask => {
+            if (this.peerconnection) {
+                thisTask.setShouldRenegotiate(
+                    this.peerconnection.removeTrackMute(track));
             }
+        });
 
-            return this.peerconnection.removeTrackMute(track);
-        };
-
-        return this._doRenegotiate('remove-as-mute', workFunction);
+        return this._doRenegotiate('remove-as-mute', queueTask);
     }
 
     /**
-     * A worker function that is scheduled and executed on the
-     * {@link modificationQueue}. Local SDPs from before and after the function
-     * is executed are compared and 'source-add'/'source-remove' notifications
-     * are being sent to remote participants (to propagate changes in local
-     * streams description).
-     * @name JingleSessionPC~WorkerFunction
-     * @function
-     * @returns {boolean|string|SDP} there are several things that this
-     * function can return:
-     * <tt>true</tt> if the renegotiation should follow the function execution
-     * <tt>false</tt> if there should be no renegotiation
-     * <tt>SDP</tt> that will be set as the remote description in the
-     * renegotiation process
-     * <tt>string</tt> which stands for the error description, no renegotiation
-     * will be done in that case it's returned
-     */
-    /**
      * Does the logic of doing the session renegotiation by updating local and
      * remote session descriptions. Will compare the local description, before
-     * and after the renegotiation to update local streams description (sends
-     * "source-add"/"source-remove" notifications).
+     * and after the {@link ModificationQueueTask} execution to update local
+     * streams description (sends "source-add"/"source-remove" notifications).
+     * See {@link ModificationQueueTask} description for more info.
      * @param {string} actionName the name of the action which will appear in
      * the events logged to the logger.
-     * @param {WorkerFunction} workFunction a function that will be executed on
-     * the queue. See type description for moe info.
+     * @param {ModificationQueueTask} modificationQueueTask the task to be
+     * executed on the queue. Check {@link ModificationQueueTask} for more info.
      * @private
      */
-    _doRenegotiate(actionName, workFunction) {
+    _doRenegotiate(actionName, modificationQueueTask) {
         const workFunctionWrap = finishedCallback => {
             // Remember localSDP from before any modifications are done, by
             // the worker function
             let oldSdp = this.peerconnection.localDescription.sdp;
-            let remoteSdp = null;
 
-            let modifySources = workFunction();
+            modificationQueueTask.execute();
+
+            const taskError = modificationQueueTask.getError();
```

Comment:
if you make the change above, this could be:
```
  if (modificationQueueTask.shouldRenegotiate) {
    // renegotiation logic
  } else if (modificationQueueTask.getError()) {
    // handle error
  }
```
which is maybe cleaner?  i dunno.  the multi-step process of calling `shouldRenegotiate` and then `getRemoteDescription` seems a little weird